composeApp/src/commonMain/kotlin/sp/bvantur/tasky/agenda/presentation/AgendaDetailsViewModel.kt

```diff
@@ -434,7 +442,31 @@ class AgendaDetailsViewModel(
         when (agendaType) {
             AgendaType.EVENT -> onSaveEvent()
             AgendaType.TASK -> onSaveTask()
-            AgendaType.REMINDER -> TODO()
+            AgendaType.REMINDER -> onSaveReminder()
+        }
+    }
+
+    private fun onSaveReminder() {
+        viewModelScope.launch {
+            agendaRepository.createReminder(
+                Reminder(
+                    reminderId = AgendaDetailsUtils.generateAgendaId(),
+                    title = viewStateFlow.value.title ?: "",
+                    description = viewStateFlow.value.description ?: "",
+                    time = viewStateFlow.value.currentFromDateTime,
+                    reminder = viewStateFlow.value.reminderValue
+                )
+            ).onError {
+                if (it.isSyncError()) {
+                    // TODO send a message that event will be saved after phone gets back internet connection
+                    emitSingleEvent(AgendaDetailsSingleEvent.CloseScreen)
+                    return@onError
```

Comment:
Syncing the agenda items to a remote server is background work - do you think this is worth sending a toast to the user about? As long as the database is up to date, the user will see their changes reflected immediately.